### PR TITLE
MOBL-486 - Add appDelegate window object check 

### PR DESCRIPTION
**Pull request by ketanshikharebsft**
For sceneDelegate configured app, appDelegate window object is not present. Add appDelegate window object check so that app should not crash if window object is not present.

### CODE REVIEW

File: BlueShift-iOS-SDK/BlueShiftDeepLink.m [hunk 1]
```diff
@@ -131,18 +136,20 @@ - (void)deepLinkToPath:(NSArray *)path {
 
 - (UIViewController *)lastViewController {
     // Get the last view controller in the view controller list ...
-    
+    if ([BlueShiftInAppNotificationHelper checkAppDelegateWindowPresent] == NO) {
+        return  nil;
```
Comment (rahulrvp) [inline BlueShift-iOS-SDK/BlueShiftDeepLink.m:140]: this nil value is handled at the caller's side, right?

File: BlueShift-iOS-SDK/InApps/BlueShiftInAppNotificationHelper.m [hunk 2]
```diff
@@ -108,6 +108,13 @@ + (CGFloat)getPresentationAreaWidth {
     return presentationAreaWidth;
 }
 
++ (BOOL)checkAppDelegateWindowPresent {
+    if (![[UIApplication sharedApplication].delegate respondsToSelector:@selector(window)]) {
+        return NO;
+    }
+    return YES;
```
Comment (rahulrvp) [inline BlueShift-iOS-SDK/InApps/BlueShiftInAppNotificationHelper.m:115]: can we inline this code like this? ignore if the syntax is not supported :) 

```return ![[UIApplication sharedApplication].delegate respondsToSelector:@selector(window)]```

File: BlueShift-iOS-SDK/BlueShiftAppDelegate.m [hunk 3]
```diff
@@ -337,12 +338,19 @@ - (void)presentInAppAlert:(NSDictionary *)userInfo {
...
     self.userInfo = userInfo;
     // Handle push notification when the app is in active state...
-    UIViewController *topViewController = [self topViewController:[[UIApplication sharedApplication].keyWindow rootViewController]];
-    BlueShiftAlertView *pushNotificationAlertView = [[BlueShiftAlertView alloc] init];
-    pushNotificationAlertView.alertControllerDelegate = (id<BlueShiftAlertControllerDelegate>)self;
-    if (@available(iOS 8.0, *)) {
-        UIAlertController *blueShiftAlertViewController = [pushNotificationAlertView alertViewWithPushDetailsDictionary:userInfo];
-        [topViewController presentViewController:blueShiftAlertViewController animated:YES completion:nil];
+    @try {
+        if ([BlueShiftInAppNotificationHelper checkAppDelegateWindowPresent] == NO) {
```
Comment (rahulrvp) [inline BlueShift-iOS-SDK/BlueShiftAppDelegate.m:342]: can we replace this with a positive statement (== YES) and bring in the below code inside this if block? will save one line of code :)

File: BlueShift-iOS-SDK/BlueShiftDeepLink.m (hunk 1 above)
Comment (ketanshikharebsft) [reply BlueShift-iOS-SDK/BlueShiftDeepLink.m:140]: Yes, it has been taken care of.

File: BlueShift-iOS-SDK/InApps/BlueShiftInAppNotificationHelper.m (hunk 2 above)
Comment (ketanshikharebsft) [reply BlueShift-iOS-SDK/InApps/BlueShiftInAppNotificationHelper.m:115]: I have kept it like this for debugging and readability point of view. Also, `[[UIApplication sharedApplication].delegate respondsToSelector:@selector(window)]` returns an object so returning it directly wont help.

File: BlueShift-iOS-SDK/BlueShiftAppDelegate.m (hunk 3 above)
Comment (ketanshikharebsft) [reply BlueShift-iOS-SDK/BlueShiftAppDelegate.m]: Fixed.

**Review by rahulrvp — APPROVED** [review]
LGTM

**Pull request merged**